src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphNode.cs

```diff
@@ -27,7 +27,7 @@ public GraphNode(LibraryRange key)
         /// <summary>
         /// Used in case that a node is removed from its outernode and needs to keep reference of its parents.
         /// </summary>
-        internal IList<GraphNode<TItem>> ParentNodes { get; }
+        public IList<GraphNode<TItem>> ParentNodes { get; }
```

Comment:
Unfortunately this has to be made public so that `LockFileBuilder` can look up the parent that brought in a centrally managed and pinned transitive dependency.